src/init.cpp

```diff
@@ -827,8 +827,10 @@ void InitParameterInteraction(ArgsManager& args)
             LogPrintf("%s: parameter interaction: -listen=0 -> setting -upnp=0\n", __func__);
         if (args.SoftSetBoolArg("-discover", false))
             LogPrintf("%s: parameter interaction: -listen=0 -> setting -discover=0\n", __func__);
-        if (args.SoftSetBoolArg("-listenonion", false))
+        if (args.GetBoolArg("-listenonion", DEFAULT_LISTEN_ONION)) {
+            args.ForceSetArg("-listenonion", std::string("0"));
             LogPrintf("%s: parameter interaction: -listen=0 -> setting -listenonion=0\n", __func__);
+        }
     }
 
```

Comment:
nit on the commit message: I think it is more common to put "fixes #20657" at the bottom of the commit message on a line of its own, rather than inside the first line. But feel free to ignore this.

Reply:
moved it to the bottom. thanks for the hint